### PR TITLE
add the canary test plan

### PR DESCRIPTION
**Pull request by kissiel**
## Description

This test plan is meant to run on Checkbox versions that are candidates for a release. The jobs included here are explicitly listed so the same jobs are  run in each session. Moreover, the jobs picked here are ones that on a stable device should all pass if the Checkbox stack is ok. In other words, failure to successfully run all the tests contained here should signify that there is a bug in the Checkbox code.

## Resolved issues
Resolves [CHECKBOX-795](https://warthogs.atlassian.net/browse/CHECKBOX-795)

[CHECKBOX-795]: https://warthogs.atlassian.net/browse/CHECKBOX-795?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pieqq — CHANGES_REQUESTED** [review]
This test plan looks good. It covers:

- normal jobs
- template jobs
- template jobs using the jinja2 engine
- different kinds of reboot
- suspend/resume
- after-suspend jobs

It doesn't have any graphics-related job though; how about adding `graphics/{index}_driver_version_{product_slug}` (it's a before and after suspend job that spits out some driver info)? It would requires `graphics_card` resource job.

I also have a question inline.

File: providers/base/units/canary/canary.pxu [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+ com.canonical.certification::memory/info
+ com.canonical.certification::ipv6_detect
+ com.canonical.certification::ipv6_link_local_address_.*
+ com.canonical.certification::networking/predictable_names
+ com.canonical.certification::power-management/warm-reboot
+ com.canonical.certification::power-management/warm-reboot
+ com.canonical.certification::power-management/post-warm-reboot
+ com.canonical.certification::power-management/cold-reboot
+ com.canonical.certification::power-management/cold-reboot
+ com.canonical.certification::power-management/post-cold-reboot
```
[the remark this answers is not in the document]
Comment (kissiel) [reply providers/base/units/canary/canary.pxu]: That's my mistake stemming from the approach I took to generate the list. Removed in 604108b27a9a2418cdc656f32582a1f18dd533b7

**Comment by kissiel** [comment]
> It doesn't have any graphics-related job though; how about adding `graphics/{index}_driver_version_{product_slug}` (it's a before and after suspend job that spits out some driver info)? It would requires `graphics_card` resource job.

Great idea. So here goes your first addition compared to what we had previously for edge testing (as it used IoT TP, so no GPU stuff). Added in d30661e1f4b3ed286dd726c77f100f0fcf6492d4.

**Review by pieqq — APPROVED** [review]
All good, thanks!

**Pull request merged**